src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,66 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunStartupBenchmark(string[] args, ProcessStartInfo psi, PreDel pre, MidDel mid, PostDel post)
+        {
+            pre(_performanceTestConfig, psi);
```

Comment:
>pre(_performanceTestConfig, psi); [](start = 12, length = 33)

I think that the Setup and Teardown delegates should be used around the benchmark iterations, and not before and after it. It should be use to keep a clean state between each benchmark iteration.